lib/compat.php

```diff
@@ -8,6 +8,45 @@
  * @package gutenberg
  */
 
+if ( ! function_exists( 'register_block_type_from_metadata' ) ) {
+	/**
+	 * Registers a block type from metadata stored in the `block.json` file.
+	 *
+	 * @since 7.8.0
+	 *
+	 * @param string $path Path to the folder where the `block.json` file is located.
+	 *                     Alternatively it's possible to provide only the name of the
+	 *                     last folder for blocks located in the WordPress core.
+	 * @param array  $args {
+	 *     Optional. Array of block type arguments. Any arguments may be defined, however the
+	 *     ones described below are supported by default. Default empty array.
+	 *
+	 *     @type callable $render_callback Callback used to render blocks of this block type.
+	 * }
+	 * @return WP_Block_Type|false The registered block type on success, or false on failure.
+	 */
+	function register_block_type_from_metadata( $path, $args = array() ) {
+		$path = is_dir( $path ) ? $path : ABSPATH . WPINC . '/blocks/' . $path;
```

Comment:
Do we really need to optimize for core blocks as a special case? Doesn't seem like we're even taking advantage of that in what we have for those core block implementations.

Reply:
> Do we really need to optimize for core blocks as a special case?

It's not strictly necessary but perhaps nice-to-have?

This started as a convenience function for core that can be used by plugins. As the path for the default blocks in wp-includes is known, it can be called without needing to pass more info. In that it is somewhat similar to how script-loader works (only the script "handle" is needed to enqueue a script). If something ever changes there, it will be enough to update this function and not worry about any back-compat.

Reply:
Let's remove this one line conditional for now and revisit it later when it's backported to the WordPress core. I don't think it can be properly tested in Gutenberg. I add it only because of the previous discussion as it seemed like a good approach from the core's perspective.